providers/writing_a_new_provider.md

```diff
@@ -0,0 +1,555 @@
+```bash
+$ git add .
+$ git commit  -m "Initial commit"
+```
+
+Now that we have both sides linked up lets verify that everything worked:
+```ruby
+$ bundle exec rails c
+>> ManageIQ::Providers::AwesomeCloud
+=> ManageIQ::Providers::AwesomeCloud
+>> ManageIQ::Providers::AwesomeCloud::CloudManager
+=> ManageIQ::Providers::AwesomeCloud::CloudManager (call 'ManageIQ::Providers::AwesomeCloud::CloudManager.connection' to establish a connection)
+```
+
+Success!  That means that core ManageIQ knows about our new cloud provider.
+
+Now lets get that provider added so we have something to play with:
+```ruby
+>> ems = ManageIQ::Providers::AwesomeCloud::CloudManager.create!(:name => "My Awesome Cloud", :zone => Zone.default_zone)
+```
+
+Now that we have that done it is time to start filling out our new provider.  The first step is to find the SDK gem for this provider.  If there isn't a provider SDK for Ruby you have a few options which we'll cover later.  For now lets assume that Awesome Cloud has a ruby gem called 'awesome_cloud'.
+
+### Add your provider's SDK to the gemspec
+
+Let's add this to our provider's gemspec:
+```bash
+$ git diff
+diff --git a/manageiq-providers-awesome_cloud.gemspec b/manageiq-providers-awesome_cloud.gemspec
+index 6c228c2..91e8e71 100644
+--- a/manageiq-providers-awesome_cloud.gemspec
++++ b/manageiq-providers-awesome_cloud.gemspec
+@@ -18,6 +18,8 @@ Gem::Specification.new do |spec|
+   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+   spec.require_paths = ["lib"]
+
++  spec.add_dependency "awesome_cloud", "~> 0.1"
++
+   spec.add_development_dependency "manageiq-style"
+   spec.add_development_dependency "simplecov"
+```
+
+Then bundle update to pull in the change
+```
+$ bundle update
+```
+
+Now that we have the gem installed we can start to write our connection code.  ManageIQ providers have to expose a `#connect` and a `#verify_credentials` method on the class and the instance.  The class method is used when adding a provider (when there is no instance record yet) and the instance methods are used after.
+
+Let's assume that Awesome Cloud requires a region, access_key, and secret_key in order to connect.
+
+```ruby
+>> ems = ManageIQ::Providers::AwesomeCloud::CloudManager.first
+>> ems.update!(:provider_region => "us-east-1")
+>> ems.update_authentication(:default => {:userid => "MY-ACCESS-KEY", :password => "MY-SECRET-KEY"})
+```
+
+Now let's add connect methods to our provider.
+
+### Connecting and verifying credentials
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::CloudManager < ManageIQ::Providers::CloudManager
+  def self.raw_connect(region, access_key, secret_key, service = "Compute")
+    require "awesome_cloud/#{service.to_s.downcase}"
+    credentials = AwesomeCloud::Credentials.new(:access_key => access_key, :secret_key => secret_key)
+
+    AwesomeCloud.const_get(service).new(:credentials => credentials, :region => region)
```

Comment:
Since it's a fake sdk, let's simplifiy

```suggestion
    AwesomeCloud.const_get(service).new(:access_key => access_key, :secret_key => secret_key, :region => region)
```